keystorage/aws/src/test/java/tech/pegasys/signers/aws/CacheableAwsSecretsManagerProviderTest.java

```diff
@@ -0,0 +1,125 @@
+class CacheableAwsSecretsManagerProviderTest {
+
+  private final String AWS_ACCESS_KEY_ID = System.getenv("AWS_ACCESS_KEY_ID");
+  private final String AWS_SECRET_ACCESS_KEY = System.getenv("AWS_SECRET_ACCESS_KEY");
+  private final String AWS_REGION = "us-east-2";
+  private final String DIFFERENT_AWS_ACCESS_KEY_ID = System.getenv("RW_AWS_ACCESS_KEY_ID");
+  private final String DIFFERENT_AWS_SECRET_ACCESS_KEY = System.getenv("RW_AWS_SECRET_ACCESS_KEY");
+  private final String DIFFERENT_AWS_REGION = "us-east-1";
+
+  private CacheableAwsSecretsManagerProvider cacheableAwsSecretsManagerProvider;
+  private long cacheMaximumSize;
+
+  private void verifyEnvironmentVariables() {
+    Assumptions.assumeTrue(
+        DIFFERENT_AWS_ACCESS_KEY_ID != null, "Set RW_AWS_ACCESS_KEY_ID environment variable");
+    Assumptions.assumeTrue(
+        DIFFERENT_AWS_SECRET_ACCESS_KEY != null,
+        "Set RW_AWS_SECRET_ACCESS_KEY environment variable");
+    Assumptions.assumeTrue(AWS_ACCESS_KEY_ID != null, "Set AWS_ACCESS_KEY_ID environment variable");
+    Assumptions.assumeTrue(
+        AWS_SECRET_ACCESS_KEY != null, "Set AWS_SECRET_ACCESS_KEY environment variable");
+  }
+
+  private void initializeCacheableAwsSecretsManagerProvider() {
+    cacheMaximumSize = 4;
+    cacheableAwsSecretsManagerProvider = new CacheableAwsSecretsManagerProvider(cacheMaximumSize);
+  }
+
+  private AwsSecretsManager createDefaultSecretsManager() {
+    return cacheableAwsSecretsManagerProvider.createAwsSecretsManager();
+  }
+
+  private AwsSecretsManager createSpecifiedSecretsManager() {
+    return cacheableAwsSecretsManagerProvider.createAwsSecretsManager(
+        AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_REGION);
+  }
+
+  private AwsSecretsManager createSecretsManagerDifferentKeys() {
+    return cacheableAwsSecretsManagerProvider.createAwsSecretsManager(
+        DIFFERENT_AWS_ACCESS_KEY_ID, DIFFERENT_AWS_SECRET_ACCESS_KEY, AWS_REGION);
+  }
+
+  private AwsSecretsManager createSecretsManagerDifferentRegion() {
+    return cacheableAwsSecretsManagerProvider.createAwsSecretsManager(
+        AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, DIFFERENT_AWS_REGION);
+  }
+
+  private AwsSecretsManager createSecretsManagerDifferentKeysDifferentRegion() {
+    return cacheableAwsSecretsManagerProvider.createAwsSecretsManager(
+        DIFFERENT_AWS_ACCESS_KEY_ID, DIFFERENT_AWS_SECRET_ACCESS_KEY, DIFFERENT_AWS_REGION);
+  }
+
+  @BeforeAll
+  void setup() {
+    verifyEnvironmentVariables();
+    initializeCacheableAwsSecretsManagerProvider();
+  }
+
+  @AfterEach
+  void teardown() {
+    cacheableAwsSecretsManagerProvider.close();
+  }
+
+  @Test
+  void isSameAsCachedSpecifiedSecretsManager() {
+    assertThat(createSpecifiedSecretsManager()).isSameAs(createSpecifiedSecretsManager());
+  }
+
+  @Test
+  void isSameAsCachedDefaultSecretsManager() {
+    assertThat(createDefaultSecretsManager()).isSameAs(createDefaultSecretsManager());
+  }
```

Comment:
How about a test like this?

```suggestion
  }
  
  @Test
  void secretsManagerIsNotCachedWhenCacheSizeIsSetToZero() {
    cacheableAwsSecretsManagerProvider = new CacheableAwsSecretsManagerProvider(0);
    assertThat(createSpecifiedSecretsManager()).isNotSameAs(createSpecifiedSecretsManager());
  }
```

Note, you'd have to move `initializeCacheableAwsSecretsManagerProvider()` instead a @BeforeEach instead of @BeforeAll for this to work